src/sif/autocost.cc

```diff
@@ -491,9 +492,10 @@ bool AutoCost::ModeSpecificAllowed(const baldr::AccessRestriction& restriction)
 Cost AutoCost::EdgeCost(const baldr::DirectedEdge* edge,
                         const graph_tile_ptr& tile,
                         const uint32_t seconds,
-                        uint8_t& flow_sources) const {
+                        uint8_t& flow_sources,
+                        const uint64_t seconds_from_now) const {
   // either the computed edge speed or optional top_speed
-  auto edge_speed = tile->GetSpeed(edge, flow_mask_, seconds, false, &flow_sources);
+  auto edge_speed = tile->GetSpeed(edge, flow_mask_, seconds, false, &flow_sources, seconds_from_now);
```

Comment:
and here the same. looks like we can replace both `seconds` and `seconds_from_now` by `time_info`

Reply:
GetSpeed can not be changed in this PR, as timeinfo object refers it through graphreader, I'll try to fix this relation in another PR. (It's strange that TimeInfo class has dependance on graph